source/features/quick-repo-deletion.tsx

```diff
@@ -22,12 +30,103 @@ async function isRepoUnpopular(): Promise<boolean> {
 	return counter!.textContent === '0';
 }
 
+async function tokenHasDeleteRepoScope(): Promise<boolean> {
+	try {
+		await expectTokenScope('delete_repo');
+		return true;
+	} catch {
+		return false;
+	}
+}
+
+async function notifyMissingTokenScope(): Promise<void> {
+	await addNotice(['Token does not have delete_repo scope.'], {
+		type: 'error',
+		action: (
+			<a className="btn btn-sm primary flash-action" href="https://github.com/settings/tokens">
+				Update token…
+			</a>
+		),
+	});
+}
+
+async function notifyDeletionFailure(error: Error): Promise<void> {
+	await addNotice([
+		'Could not delete the repository. ',
+		(error as RefinedGitHubAPIError).response?.message ?? error.message,
+	], {
+		type: 'error',
+	});
+}
+
+async function deleteRepository(nameWithOwner: string): Promise<void> {
+	await api.v3('/repos/' + nameWithOwner, {
+		method: 'DELETE',
+		json: false,
+	});
+}
+
+async function modifyUIAfterSuccessfulDeletion(): Promise<void> {
+	const {nameWithOwner, owner} = getRepo()!;
+	const forkSource = '/' + getForkedRepo()!;
+	const restoreURL = pageDetect.isOrganizationRepo()
+		? `/organizations/${owner}/settings/deleted_repositories`
+		: '/settings/deleted_repositories';
+	const otherForksURL = `/${owner}?tab=repositories&type=fork`;
+
+	await addNotice(
+		<>
+			<TrashIcon />
+			<span>
+				Repository <strong>{nameWithOwner}</strong> deleted. <a href={restoreURL}>Restore it</a>, <a href={forkSource}>visit the source repo</a>, or see <a href={otherForksURL}>your other forks.</a>
+			</span>
+		</>,
+		{action: false},
+	);
+	$('.application-main')!.remove();
+}
+
+async function performDeletion(): Promise<void> {
+	if (!(await tokenHasDeleteRepoScope())) {
```

Comment:
```suggestion
	// TODO: Add support for app tokens 
	if (!(await tokenHasDeleteRepoScope())) {
```

I think app tokens don't have the same concept of scopes, so they would fail even if the token is very capable. See the last paragraph on https://github.com/refined-github/refined-github/wiki/Security#token